src/prefect/tasks/snowflake/snowflake.py

```diff
@@ -47,51 +48,91 @@ def __init__(
         self.account = account
         self.user = user
         self.password = password
+        self.private_key = private_key
         self.database = database
         self.schema = schema
         self.role = role
         self.warehouse = warehouse
         self.query = query
         self.data = data
         self.autocommit = autocommit
-        self.private_key = private_key
         super().__init__(**kwargs)
 
-    @defaults_from_attrs("query", "data", "autocommit")
-    def run(self, query: str = None, data: tuple = None, autocommit: bool = None):
+    @defaults_from_attrs(
```

Comment:
should we also include `warehouse` here? (it might not make sense, I'm not very familiar with Snowflake)